kani-driver/src/metadata.rs

```diff
@@ -110,7 +120,7 @@ impl KaniSession {
     pub fn collect_kani_metadata(&self, files: &[PathBuf]) -> Result<KaniMetadata> {
         if self.args.dry_run {
             // Mock an answer
-            Ok(KaniMetadata { proof_harnesses: vec![] })
+            Ok(KaniMetadata { proof_harnesses: vec![generate_mock_harness()] })
```

Comment:
@danielsn has previously proposed that `--dry-run` should actually do the build still, so we actually get the real proof harness metadata.

Basically, so you can run the cbmc command manually with some tweaks or whatever during development.

IMO if we're changing dry-run maybe that's what we actually change it to?

Reply:
We could add dry-run to the compiler and only collect the metadata.

Reply:
The concept of dry does get a bit fuzzy though if we now run some commands but not others. I think I'm that case you probably want to have a not so verbose mode that prints the command before running. Like cargo does